nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/SolrUtils.java

```diff
@@ -463,6 +463,8 @@ private static void writeValue(final SolrInputDocument inputDocument, final Obje
                     addFieldToSolrDocument(inputDocument,fieldName,(BigInteger)coercedValue,fieldsToIndex);
                 }
                 break;
+            case BIGDECIMAL:
+                addFieldToSolrDocument(inputDocument, fieldName, coercedValue, fieldsToIndex);

```

Comment:
I'd use `DataTypeUtils.toBigDecimal(coercedValue, fieldName)` to be safe and follow conventions.

Reply:
Sadly In a number of places I followed the BigInt approach, which missed some of these safe(r) handling.